main.nf

```diff
@@ -281,7 +283,7 @@ process BWA {
     
     output:
         file "${sample_id}.log"
-        tuple val(sample_id), path(bam), 
+        tuple val('bwa'), val(sample_id), path(bam), 
               val(align_dir), emit: 'align'
     
     script:
```

Comment:
Please:

-  refactor out the `bwa index` into a separate process so that the reference fasta index can be shared between samples. 
- remove `-P` flag from `bwa mem` command
- pipe stdout from `bwa mem` straight to `samtools sort` (e.g. `bwa mem $ref $r1 $r2 | samtools sort | samtools view -F4 > $bam`) 
- `bwa mem` typically doesn't produce any log file. I think only `bwa index` does.

Reply:
The bwa mem flag -P has been removed and the pipe is implemented without the align file. logging is still in place and will be removed once bwa index is in a separate process.